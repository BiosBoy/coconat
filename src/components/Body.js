import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

import styles from '../styles/index.scss';
import buttonsStyles from '../styles/button.scss';
import body from '../styles/body.scss';

const FIRST_IMAGE_ID = 1;
const SECOND_IMAGE_ID = 2;

class Body extends PureComponent {
  static defaultProps = {
    imageToShow: 3,
    switchImage: () => {}
  }

  static propTypes = {
    imageToShow: PropTypes.number,
    switchImage: PropTypes.func
  }

  _handleClick = () => {
    const { imageToShow, switchImage } = this.props;

    if (imageToShow === FIRST_IMAGE_ID) {
      return switchImage(SECOND_IMAGE_ID);
    }

    return switchImage(FIRST_IMAGE_ID);
  }

  _getImage = () => {
    const { imageToShow } = this.props;

    return imageToShow;
  }

  render() {
    const imageToShow = this._getImage();

    return (
      <div className={styles.body}>
        <button type='button' onClick={this._handleClick} className={buttonsStyles.button}>
          <TransitionGroup className={body.animWrap}>
            <CSSTransition classNames='mainImage' timeout={500} key={imageToShow}>
              <img className={styles.bodyImg} src={`../assets/${imageToShow}.png`} alt='main_img' />
            </CSSTransition>
          </TransitionGroup>
        </button>
      </div>
    );
  }
}

export default Body;
