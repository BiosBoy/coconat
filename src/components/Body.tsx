import React, { memo, useCallback } from 'react';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

import styles from '../styles/index.scss';
import buttonsStyles from '../styles/button.scss';
import body from '../styles/body.scss';

const FIRST_IMAGE_ID = 1;
const SECOND_IMAGE_ID = 2;

interface IBody {
  imageToShow: number
  switchImage: (imgNumber: number) => void
}

const Body = memo((props: IBody) => {
  const { imageToShow, switchImage } = props;

  const handlerClick = useCallback(
    () => {
      if (imageToShow === FIRST_IMAGE_ID) {
        return switchImage(SECOND_IMAGE_ID);
      }

      return switchImage(FIRST_IMAGE_ID);
    },
    [imageToShow, switchImage]
  );

  return (
    <div className={styles.body}>
      <button type='button' onClick={handlerClick} className={buttonsStyles.button}>
        <TransitionGroup className={body.animWrap}>
          <CSSTransition classNames='mainImage' timeout={500} key={imageToShow}>
            <img className={styles.bodyImg} src={`../assets/${imageToShow}.png`} alt='main_img' />
          </CSSTransition>
        </TransitionGroup>
      </button>
    </div>
  );
});

Body.displayName = 'Body';

export default Body;
