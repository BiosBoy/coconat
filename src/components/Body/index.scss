.animWrap {
  position: relative;
}

.textLabel {
  color: #111;
}

.red {
  background-color: red;
}

.black {
  background-color: black;
}

.transparent {
  background-color: transparent;
}

:global {
  .mainImage-enter {
    opacity: 0;
    position: absolute;
    transition: all 500ms linear;
    transform: translateX(-10%);
  }

  .mainImage-enter-active {
    opacity: 1;
    transform: translateX(0%);
  }

  .mainImage-exit {
    transition: all 500ms linear;
    transform: translateX(0%);
    opacity: 1;
  }

  .mainImage-exit-active {
    transition: all 500ms linear;
    transform: translateX(10%);
    opacity: 0;
  }
}
