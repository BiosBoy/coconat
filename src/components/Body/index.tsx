import React, { memo } from 'react'

import './index.scss'

const Body = memo(() => {
  return <div>Hello World!</div>
})

Body.displayName = 'Body'

export default Body
