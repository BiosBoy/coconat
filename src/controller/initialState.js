const initialState = {
  common: {
    appName: 'react-suspense'
  }
};

export default initialState;
