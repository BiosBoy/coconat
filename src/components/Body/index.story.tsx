import React from 'react';
import { select, boolean, text } from '@storybook/addon-knobs';

import Body, { IBody } from './';

export const Default = () => {
  const config: IBody = {
    backgroundColor: select('Select background color', ['red', 'black', 'transparent'], 'transparent'),
    withLabel: boolean('Activate Label:', false),
    labelText: text('Set Label text:', ''),
    imageToShow: 1,
    switchImage: () => {}
  }

  return <Body {...config} />
}

export default {
  title: 'Button'
}
