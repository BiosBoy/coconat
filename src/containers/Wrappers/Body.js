import { connect } from 'react-redux';
import { Body } from '../../components';
import { switchImage } from '../../modules/actions';

const mapStateToProps = state => ({
  imageToShow: state.app.imageToShow
});

const mapDispatchToProps = dispatch => ({
  switchImage: imageID => dispatch(switchImage(imageID))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Body);
