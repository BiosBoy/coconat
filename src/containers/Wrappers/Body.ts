import { connect } from 'react-redux';
import { Body } from '../../components';
import { switchImage } from '../../modules/actions';

interface ISwitchImage {
  type: string
  imageID: number
}

interface IStore {
  app: {
    imageToShow: number
  }
  dispatch: (someFunc: ISwitchImage) => void
}

const mapStateToProps = (state: IStore) => ({
  imageToShow: state.app.imageToShow
});

const mapDispatchToProps = (dispatch: IStore['dispatch']) => ({
  switchImage: (imageID: ISwitchImage['imageID']) => dispatch(switchImage(imageID))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Body);
