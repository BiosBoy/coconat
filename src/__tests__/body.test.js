import React from 'react';
import { mount } from 'enzyme';

import initialState from './mocks/body';

import Body from '../components/Body';

describe('<Body Component Test />', () => {
  it('should render basic Body Component with an image', () => {
    const Component = mount(<Body {...initialState} />);

    expect(Component.find('.body').length).toBe(1);
    expect(Component.find('.bodyImg').length).toBe(1);

    expect(Component).toMatchSnapshot();
  });
});
