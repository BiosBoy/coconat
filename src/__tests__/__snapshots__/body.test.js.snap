// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body Component Test /> should render basic Body Component with an image 1`] = `
<Body
  imageToShow={3}
  switchImage={[Function]}
>
  <div
    className="body"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      <TransitionGroup
        childFactory={[Function]}
        className="animWrap"
        component="div"
      >
        <div
          className="animWrap"
        >
          <CSSTransition
            classNames="mainImage"
            in={true}
            key=".$3"
            onExited={[Function]}
            timeout={500}
          >
            <Transition
              appear={false}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={500}
              unmountOnExit={false}
            >
              <img
                alt="main_img"
                className="bodyImg"
                src="../assets/3.png"
              />
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </button>
  </div>
</Body>
`;
